test/mod/test_meta.py

```diff
@@ -1,7 +1,165 @@
+import os
+from tempfile import TemporaryDirectory
+
+import pytest
+
 import osbuild
 import osbuild.meta
 
 
+@pytest.fixture(name="tempdir")
+def tempdir_fixture():
+    with TemporaryDirectory(prefix="meta-") as tmp:
+        yield tmp
+
+
+def create_runners(path, distro: str, base: str, versions):
+    """Create runner"""
+    runners = []
+
+    basename = f"{distro}{base}"
+    basepath = os.path.join(path, basename)
+    with open(basepath, "w+", encoding="utf-8") as f:
+        f.write("#!/bin/bash")
+        runners.append(basename)
+
+    for x in sorted(versions):
+        name = f"{distro}{x}"
+        link = os.path.join(path, name)
+        os.symlink(basepath, link)
+        runners.append(name)
+
+    return runners
+
+
+def test_runner_info():
+    table = {
+        "arch": {
+            "distro": "arch",
+            "version": 0,
+        },
+        "fedora30": {
+            "distro": "fedora",
+            "version": 30,
+        },
+        "rhel7": {
+            "distro": "rhel",
+            "version": 7,
+        },
+        "ubuntu1804": {
+            "distro": "ubuntu",
+            "version": 1804,
+        }
+    }
+
+    for name, want in table.items():
+        d, v = osbuild.meta.RunnerInfo.parse_name(name)
+
+        assert d == want["distro"]
+        assert v == want["version"]
+
+
+def test_runner_detection(tempdir):
+
+    runners = os.path.join(tempdir, "runners")
+    os.makedirs(runners)
+
+    meta = osbuild.meta.Index(tempdir)
+
+    table = {
+        "arch": {
+            "base": "",
+            "versions": [],
+            "check": {"": 0},
+            "fail": []
+        },
+        "fedora": {
+            "base": 30,
+            "versions": list(range(31, 40)),
+            "check": {40: 39, 31: 31, 35: 35},
+            "fail": [29],
+        },
+        "ubuntu": {
+            "base": 1810,
+            "versions": [1904, 1910, 2004],
+            "check": {2010: 2004, 1912: 1910},
+            "fail": [1804],
+        },
+        "rhel": {
+            "base": 90,
+            "versions":  [91, 92, 93],
+            "check": {94: 93},
+        },
+        "future": {
+            "base": 100,
+            "versions":  [101, 102, 103],
+            "check": {110: 103},
+        }
+    }
+
+    want_all = []
+    for distro, info in table.items():
+        base = info["base"] or 0
+        versions = info["versions"]
+        want = create_runners(runners, distro, str(base), versions)
+        have = meta.list_runners(distro)
```

Comment:
You do not re-create the `tmpdir` so this keeps adding more and more runners and re-iterating the dir in `list_runners()`. I don't mind it much, but it will fall over once we add caching in `list_runners()`, or similar. It is shorter now, so I guess lets keep it. I just wondered about it when reading.

Reply:
Right, I did not read till the end. In a later patch you re-create the index on each run. Ok!